dsl/interpreter.py

```diff
@@ -0,0 +1,164 @@
+import os
+import re
+import logging
+
+from dsl.parser import Grammar
+
+logger = logging.getLogger('example_logger')
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+DSL_EXTENSION = '.str'
+MODULES_DIR_NAME = 'dsl/modules'
+modules = {}
+def load_modules():
+    content = os.listdir(MODULES_DIR_NAME)
+    for file in content:
+        if os.path.isfile(os.path.join(MODULES_DIR_NAME, file)) and file.endswith('.py'):
+            name = file[:len(file)-3]
+            modules[name] = exec()
+
+
+def parse_file(input_name: str) -> str | None:
```

Comment:
I'd switched to pathlib: a high-level library for paths
```suggestion
def parse_file(input_name: Path) -> Path | None:
```